### PR TITLE
chore: refactor core package

### LINKED ISSUE #115: Build diagram showing how charmil validator works
Base off documentation for validator

### LINKED ISSUE #183: Cleanup Core package
### Feature or problem description

I have reviewed core package and it looks like what we have there is mostly non functional, undocumented or outdated.
Removing entire core at this point will result in no los in the value.

What we should do:

- Review every folder we have there
- Remove factory (as we agreed that factory is starter artifact not core artifact)
- Fix config - config is key for us to have proper modular architecture but provided solution doesn't seem to work.

### LINKED ISSUE #213: Core should have folders representing different roles
- utils
- commands (will be empty right now but can land completions)
- config

### PR DESCRIPTION
**Pull request by ankithans**
Closes #213 
Closes #183
Closes #115 

### Description
- refactor core structure into folders
- move factory from internal -> pkg
- remove charmil commands docs(outdated)
- update factory, i18n docs

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation change
- [ ] Other (please specify)

### Checklist

- [x] Documentation added for the feature
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: docs/src/charmil_factory.md [hunk 1]
```diff
@@ -3,38 +3,49 @@ title: Charmil Factory
...
+defaultLocales embed.FS
+
+// create a config handler
+h := config.NewHandler("./config.json", cfg)
+
+// Loads config values from the local config file
+err := h.Load()
+if err != nil {
+    log.Fatal(err)
 }
```
Comment (namit-chandwani) [inline docs/src/charmil_factory.md:30]:
```suggestion
```
This part can be removed as it's independent and has nothing to do with the factory initialization.

**Comment by namit-chandwani** [comment]
@ankithans I believe now you'll be making similar changes in the `charmil-starter` repo too

**Review by namit-chandwani — APPROVED** [review]

**Comment by ankithans** [comment]
yes after release...

**Comment by wtrocki** [comment]
>  I believe now you'll be making similar changes in the charmil-starter repo too

No changes to charmil starter. Let's build script that will do something like this:

```
mkdir newrepo
cd newrepo
git init 
git remote add starter ...
cp -Rf ../starter ./
git commit -a -m"Starter update"
git push origin main
```

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
Do you mind publishing website (manually for now)

**Comment by ankithans** [comment]
done

**Pull request merged**